runtime-tools-quarkus-extension-parent/runtime/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/forms/impl/FormsStorageImpl.java

```diff
@@ -37,39 +54,120 @@
+        } else {
+            matchedNames = formBaseInfoCollection.keySet();
+        }
 
-        forms.forEach(form -> {
-            System.out.println(form.getName());
+        matchedNames.forEach(name -> {
+            Collection<FormInfo> formBaseInfos = formBaseInfoCollection.get(name);
+            formBaseInfos.forEach(f -> {
+                if (!f.getType().equals(FormInfo.FormType.CONFIG)) {
+                    formInfoCollection.add(f);
+                }
+            });
         });
+        return formInfoCollection;
+    }
 
-        return forms.size();
+    private FormInfo.FormType getFormType(String type) {
+        switch (type) {
+            case "html":
+                return FormInfo.FormType.HTML;
+            case "tsx":
+                return FormInfo.FormType.TSX;
+            case "config":
+                return FormInfo.FormType.CONFIG;
+        }
+        return null;
     }
 
     @Override
-    public Collection<FormInfo> getFormList() {
-        return readForms().stream()
-                .map(file -> new FormInfo(FilenameUtils.removeExtension(file.getName()), FilenameUtils.getExtension(file.getName()), new Date(file.lastModified())))
-                .collect(Collectors.toList());
+    public Form getFormContent(String formName) throws IOException {
+        File formFile = getFormFile(formName);
+        File formConfig = getFormConfigFile(formName);
+        FormConfiguration formConfiguration = null;
+        if (formConfig != null && formConfig.exists()) {
+            formConfiguration = new FormConfiguration(IOUtils.toString(new FileInputStream(formConfig), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0} dose not found");
+        }
+        Form form = new Form(formConfiguration);
+        if (formFile != null && formFile.exists()) {
+            form.setSource(IOUtils.toString(new FileInputStream(formFile), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0}'s config file dose not found");
+        }
+        return form;
     }
 
-    private Collection<File> readForms() {
-        File rootFolder = FileUtils.toFile(formsFolder);
+    private File getFormFile(String formName) {
+        Collection<FormInfo> formBaseInfo = formBaseInfoCollection.get(formName);
+        if (formBaseInfo != null) {
+            Optional<FormInfo> optional = formBaseInfo.stream().filter(formBase -> !formBase.getType().getValue().equals("config")).findFirst();
 
-        return FileUtils.listFiles(rootFolder, new String[] { "html", "tsx" }, false);
+            if (optional.isPresent()) {
+                return optional.get().getFile();
+            }
+        }
+        return null;
+    }
 
+    private File getFormConfigFile(String formName) {
+        Collection<FormInfo> formBaseInfo = formBaseInfoCollection.get(formName);
```

Comment:
it should be only one FormInfo, not a collection. From that form Info we may know the we can get the .config file